lighthouse-core/test/audits/treemap-data-test.js

```diff
@@ -0,0 +1,335 @@
+      const mainUrl = 'https://squoosh.app';
+      const scriptUrl = 'https://squoosh.app/main-app.js';
+      const networkRecords = [generateRecord(scriptUrl, content.length, 'Script')];
+
+      // Add a script with no source map or usage.
+      const noSourceMapScript = {src: 'https://sqoosh.app/no-map-or-usage.js', content: '// hi'};
+      networkRecords.push(
+        generateRecord(noSourceMapScript.src, noSourceMapScript.content.length, 'Script')
+      );
+
+      const artifacts = {
+        URL: {requestedUrl: mainUrl, finalUrl: mainUrl},
+        JsUsage: {[usage.url]: [usage]},
+        devtoolsLogs: {defaultPass: networkRecordsToDevtoolsLog(networkRecords)},
+        SourceMaps: [{scriptUrl: scriptUrl, map}],
+        ScriptElements: [{src: scriptUrl, content}, noSourceMapScript],
+      };
+      const results = await TreemapData.audit(artifacts, context);
+
+      // @ts-expect-error: Debug data.
+      treemapData = results.details.treemapData;
+    });
+
+    it('basics', () => {
+      expect(Object.keys(treemapData)).toEqual(['scripts', 'resources']);
+    });
+
+    it('scripts', () => {
+      expect(treemapData.scripts.find(s => s.name === 'https://sqoosh.app/no-map-or-usage.js'))
+        .toMatchInlineSnapshot(`
+        Object {
+          "name": "https://sqoosh.app/no-map-or-usage.js",
+          "node": Object {
+            "name": "https://sqoosh.app/no-map-or-usage.js",
+            "resourceBytes": 37,
+          },
+        }
+      `);
+
+      expect(treemapData.scripts).toMatchSnapshot();
+    });
+
+    it('resources', () => {
+      const scriptsNode = (treemapData.resources[0].node.children || [])[0];
+      expect(scriptsNode.children || []).toHaveLength(2);
+
+      expect(treemapData.resources).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "name": "Resource Summary",
+            "node": Object {
+              "children": Array [
+                Object {
+                  "children": Array [
+                    Object {
+                      "name": "https://squoosh.app/main-app.js",
+                      "resourceBytes": 83748,
+                    },
+                    Object {
+                      "name": "https://sqoosh.app/no-map-or-usage.js",
+                      "resourceBytes": 5,
+                    },
+                  ],
+                  "name": "script",
+                  "resourceBytes": 83753,
+                },
+              ],
+              "name": "2 requests",
+              "resourceBytes": 83753,
+            },
+          },
+        ]
+      `);
+    });
+  });
+
+  describe('.prepareTreemapNodes', () => {
+    it('uses node data when available', () => {
+      const rootNode = TreemapData.prepareTreemapNodes('', {
+        'a.js': {resourceBytes: 100},
+        'b.js': {resourceBytes: 100, duplicate: 'blah'},
+        'c.js': {resourceBytes: 100, unusedBytes: 50},
+      });
+      expect(rootNode).toMatchInlineSnapshot(`
```

Comment:
for these cases now that they're not expected to change, can we convert to `toEqual` plus find/replace `s/Object|Array//g`

Reply:
do we work for jest, or does jest work for us? :)

Reply:
that's a good one :)

Seriously though this is more about the signal that it sends to future readers. Snapshots are generally for things that are expected to change and there's not really a "correct" answer, just "the current" answer. In all of these cases, it seems like this is actually *the* answer. If you support this sort of test where we're summing up some data by URLs as a snapshot then I'm not sure where the line is at all, like why bother having *any* assertion in jest other than a snapshot?

If your concern is about adding new data to this and having to go back to update old tests then is `toMatchObject` or `toEqual` on a subset a reasonable compromise?

If you more broadly don't like the idea of snapshots  being restricted to logic that doesn't have a correct answer, a broader proposal for why snapshots are appropriate everywhere might be worth discussing separately.